### PR TITLE
feat(Last.fm): create extension

### PR DESCRIPTION
**Pull request by eggsy**
## Description

Must extension to display **Top Songs**, **Recent Songs**, **Top & Recent Songs**, **Top Albums** and **Top Artists** using [Last.fm](https://last.fm)'s API.

## Type of change

- New extension

## Screencast

https://user-images.githubusercontent.com/13917975/149525948-9b23ae6c-ed08-4fb7-99c5-c8376f40d59f.mov

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)

### CODE REVIEW

**Comment by eggsy** [comment]
> Also updated the typo/grammar issue in Must extension's description

File: extensions/lastfm/src/albums.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    showToast(ToastStyle.Failure, "Something went wrong.", String(error));
+  }
+
+  return (
+    <List isLoading={loading} searchBarPlaceholder="Search songs...">
+      <List.Section title="Results">
+        {albums.map((a, idx) => {
+          const album = a as Album;
+          const image =
+            album.image?.find((image) => image.size === "large")?.["#text"] || "https://i.imgur.com/ekASTF4.jpg";
```
Comment (pernielsentikaer) [inline extensions/lastfm/src/albums.tsx:33]: The fallback image should be added as an asset instead

File: extensions/lastfm/src/hooks/useLastfm.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    const [error, setError] = useState<any>(null);
+
+    const handleError = (err: unknown) => {
+      setError(err);
+      setLoading(false);
+    };
+
+    const fetcher = (url: string) => fetch(url).then((r) => r.json() as Promise<SongResponse>);
+    useSWR(
+      `http://ws.audioscrobbler.com/2.0/?method=user.${method}&user=${props.username}&api_key=${
```
Comment (pernielsentikaer) [inline extensions/lastfm/src/hooks/useLastfm.ts:34]:
```suggestion
      `https://ws.audioscrobbler.com/2.0/?method=user.${method}&user=${props.username}&api_key=${
```

File: extensions/lastfm/src/hooks/useTopAlbums.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+const useTopAlbums = (props: Props) => {
+  try {
+    const [loading, setLoading] = useState(true);
+    const [error, setError] = useState<any>(null);
+    const [albums, setAlbums] = useState<Album[]>([]);
+
+    const fetcher = (url: string) => fetch(url).then((r) => r.json() as Promise<AlbumResponse>);
+    useSWR(
+      `http://ws.audioscrobbler.com/2.0/?method=user.gettopalbums&user=${props.username}&api_key=${
```
Comment (pernielsentikaer) [inline extensions/lastfm/src/hooks/useTopAlbums.ts:26]:
```suggestion
      `https://ws.audioscrobbler.com/2.0/?method=user.gettopalbums&user=${props.username}&api_key=${
```

File: extensions/lastfm/src/hooks/useTopArtists.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+const useTopArtists = (props: Props) => {
+  try {
+    const [loading, setLoading] = useState(true);
+    const [error, setError] = useState<any>(null);
+    const [artists, setArtists] = useState<Artist[]>([]);
+
+    const fetcher = (url: string) => fetch(url).then((r) => r.json() as Promise<ArtistResponse>);
+    useSWR(
+      `http://ws.audioscrobbler.com/2.0/?method=user.gettopartists&user=${props.username}&api_key=${
```
Comment (pernielsentikaer) [inline extensions/lastfm/src/hooks/useTopArtists.ts:26]:
```suggestion
      `https://ws.audioscrobbler.com/2.0/?method=user.gettopartists&user=${props.username}&api_key=${
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @eggsy 👋 

I left a few things for you to address, also, could you move the  `types` folder inside `src`.

It's also a bad idea to change other folders within another PR, in this case I will allow it, but next time we need to have it separated.

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by eggsy** [comment]
All right, created a new branch for the Must update, opened another PR for it and reverted all the changes from this branch so there will not be any confusion here!

**Review by pernielsentikaer — APPROVED** [review]
Hi @eggsy 👋 

Thanks for the update and thanks for seperating the two extensions ❤️ 

I'm going to accept it and then will my colleague do a final review before it's finally merged.

**Review by mattisssa — APPROVED** [review]

**Pull request merged**